driver-core/src/test/functional/com/mongodb/client/model/OperationTest.java

```diff
@@ -114,7 +114,6 @@ protected void assertResults(final List<Bson> pipeline, final String expectedRes
     protected List<Object> aggregateWithWindowFields(@Nullable final Object partitionBy, final WindowOutputField output) {
         List<Bson> stages = new ArrayList<>();
         stages.add(setWindowFields(partitionBy, null, output));
-        stages.add(sort(ascending("num1")));
```

Comment:
The sorting is important here:

- The order of documents produced by `$setWindowFields` is not guaranteed, even if `sortBy` for `$setWindowFields` is specified, as it is used only to within the stage.
- The test compares two lists, i.e., the test relies on the order.

Clearly, sorting by `num1` can't be hardcoded in a method that is located in `OperationTest`, but this method may accept a `@Nullable Bson` sort specification, allowing the test `shouldSetWindowFieldWithQuantile` to specify the sorting it needs.

P.S. Note that this is not a problem in `AggregatesTest.shouldGroupWithQuantile`, because there you filter the output to get the single document you need by its `_id`.